src/Beacon.jsx

```diff
@@ -242,6 +247,50 @@ export class Beacon extends Component {
     }
   }
 
+  loadHash(persistent) {
+    const { tooltipText } = this.props;
+    const text = typeof(tooltipText) === 'string' ? tooltipText : ReactDOMServer.renderToStaticMarkup(tooltipText);
```

Comment:
interesting, yet I am afraid of breaking this in next versions of React. `renderToStaticMarkup` is server specific so it might potentially use some features which are not in the browser.